client/src/controllers/TagController.ts

```diff
@@ -20,17 +21,31 @@ export class TagController extends Controller {
   static values = {
     options: { default: {}, type: Object },
     url: String,
+    debounce: { type: Number, default: 400 },
   };
 
   declare optionsValue: any;
   declare urlValue: any;
+  declare debounceValue: number;
   tagit?: JQuery<HTMLElement>;
 
   connect() {
     const preprocessTag = this.cleanTag.bind(this);
+    // Debounces by specified debounceValue to reduce unnecessary calls to database while user types, default is 300
```

Comment:
I would recommend we remove this comment, it's better to just look where the default is set instead of re-writing what the default is.

As for below, let's follow the patterns in the other controllers where we abstract the method out and create a debounced wrapper around it in the connect method.


```
  this.autocomplete = debounce(this.autocomplete).bind(this);
```

Reply:
Noted